testng-core/src/main/java/org/testng/TestRunner.java

```diff
@@ -931,10 +932,15 @@ private void logStart() {
    *     finish
    */
   private void fireEvent(boolean isStart) {
-    for (ITestListener itl : m_testListeners) {
-      if (isStart) {
+    if (isStart) {
+      for (ITestListener itl : m_testListeners) {
         itl.onStart(this);
-      } else {
+      }
+
+    } else {
+      List<ITestListener> testListenersReversed = new ArrayList<>(m_testListeners);
+      Collections.reverse(testListenersReversed);
+      for (ITestListener itl : testListenersReversed) {
```

Comment:
Here as well, we can use the newly added method `Lists.newReversedArrayList()`

Reply:
Fixed